shoov/modules/custom/shoov_screenshot/shoov_screenshot.module

```diff
@@ -126,3 +126,53 @@ function shoov_screenshot_add_tag_to_vocabulary($term_name, $vocabulary_id) {
 
   return $tid;
 }
+
+/**
+ * Create a hash code from files content.
+ *
+ * @param $files
+ *  List of uploaded files info.
+ * @param $ui_build_id
+ *  The UI Build node ID.
+ * @param $uid
+ *  The user ID.
+ *
+ * @return string
+ *  Returns the hash code.
+ */
+function shoov_screenshot_create_hash($files, $ui_build_id, $uid) {
+  $content = '';
+
+  foreach ($files as $file) {
+    $image = file_load($file['id']);
+    $content .= $file['label'] . ':' . file_get_contents($image->uri) . ';';
```

Comment:
you don't really need the content to be human readable it can be:

```php
$hash = array();
foreach ($files as $file) {
  $image = file_load($file['id']);
  $hash[] = md5(file_get_contents($image->uri));
}

$hash[] = $ui_build_id;

$hash = md5(join(';', $hash));
```

Also, no need to check for the user ID, as it's ok if multiple people will run it - they still shouldn't create duplicates